### PR TITLE
feat: add 7702 methods to ledger

### PR DESCRIPTION
**Pull request by montelaidev**
This PR adds the 7702 methods to the ledger keyring. The method itself is not usable until the dmk support has been added. 

## Examples

---

> [!NOTE]
> **Medium Risk**
> Changes hardware signing and signature recovery for personal_sign/typed data; EIP-7702 is new delegation signing surface, though iframe/mobile paths are not live yet.
> 
> **Overview**
> Adds **EIP-7702 delegation authorization signing** to the Ledger Ethereum keyring and wires it through the V2 account API.
> 
> `LedgerKeyring.signEip7702Authorization` unlocks the account, calls the bridge’s new `deviceSignDelegationAuthorization` (chain id, contract address without `0x`, nonce), builds a signature with **yParity** (`00`/`01`) instead of legacy `27`/`28`, and verifies with `recoverEIP7702Authorization`. The `LedgerBridge` interface gains delegation types and `deviceSignDelegationAuthorization`; **iframe and mobile bridges throw** until DMK/device support exists.
> 
> **personal_sign** and **typed data** now parse Ledger’s `v` via `#parseLedgerRecoveryParam` (decimal, bare hex like `'1b'`, `0x`/`0X` hex, numbers) instead of `parseInt(..., 10)`, fixing wrong recovery when the device returns hex `v`.
> 
> V2 `LedgerKeyring` advertises `EthKeyringMethod.SignEip7702Authorization` on EOA accounts. Changelog and Jest coverage thresholds are updated; extensive unit tests cover 7702 signing and `v` parsing edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c91b13d32b8241be5f695a2d0c5048ee4b005e03. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts [hunk 1]
```diff
@@ -720,6 +772,26 @@ export class LedgerKeyring implements Keyring {
...
+    if (typeof recoveryParam === 'number') {
+      return recoveryParam;
+    }
+
+    const value = String(recoveryParam).trim();
+    const withoutPrefix =
+      value.startsWith('0x') || value.startsWith('0X') ? value.slice(2) : value;
+    const radix = /[a-f]/iu.test(withoutPrefix) ? 16 : 10;
+
+    return parseInt(withoutPrefix, radix);
```
Comment (ccharly) [inline packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts]: Actually I think the logic is wrong here.

If we have a number like `0x100` (= 256), then this would be parsed as `100` (radix=10) which is wrong.

IMO we can do without the regex too:

```suggestion
    const value = String(recoveryParam).trim();
    
    if (value.startsWith('0x') || value.startsWith('0X')) {
      return parseInt(value.slice(2), 16);
    }
    
    return parseInt(value, 10);
```

My suggestion is not fully safe either... If the input is `0x` or `0X`, then `parseInt` will return `NaN`. I'm not sure what we should do here? Cause initially we were not even filtering those `NaN` 😅 I guess we could have an `assert(result !== NaN)` too? 🤔

File: packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts [hunk 2]
```diff
@@ -593,6 +594,57 @@ export class LedgerKeyring implements Keyring {
...
+        'Ledger: Unknown error while signing EIP-7702 authorization',
+      );
+    }
+
+    const recoveryValue = this.#parseLedgerRecoveryParam(payload.v);
+    const yParity =
+      recoveryValue === 0 || recoveryValue === 1
+        ? recoveryValue
+        : recoveryValue - 27;
+    const signature = `0x${payload.r}${payload.s}${yParity.toString(16).padStart(2, '0')}`;
```
Comment (ccharly) [inline packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts]: Would be nice to have some comments here.

Also, I wouldn't compute the hex within the format string. Maybe we could have a `toSomething(recoveryValue)` that would do the right thing? Including the padding formatting?

File: packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts [hunk 3]
```diff
@@ -593,6 +594,57 @@ export class LedgerKeyring implements Keyring {
...
+  async signEip7702Authorization(
+    withAccount: Hex,
+    authorization: [chainId: number, contractAddress: Hex, nonce: number],
+  ): Promise<string> {
+    const [chainId, contractAddress, nonce] = authorization;
+    const hdPath = await this.unlockAccountByAddress(withAccount);
+
+    if (!hdPath) {
+      throw new Error(
+        'Ledger: Unknown error while signing EIP-7702 authorization',
```
Comment (gantunesr) [inline packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts]: Should this reflect that the error is related to the missing `hdPath`?

```suggestion
        'Ledger: Missing hdPath while signing EIP-7702 authorization',
```

File: packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts [hunk 4]
```diff
@@ -593,6 +594,57 @@ export class LedgerKeyring implements Keyring {
...
+        hdPath,
+        chainId,
+        contractAddress: remove0x(contractAddress),
+        nonce,
+      });
+    } catch (error: unknown) {
+      handleLedgerTransportError(
+        error,
+        'Ledger: Unknown error while signing EIP-7702 authorization',
+      );
```
Comment (gantunesr) [inline packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts:624]: I think we can also remove the `Unknown` here since we have the actual `error` as the first param for the handler

```suggestion
    } catch (error: unknown) {
      handleLedgerTransportError(
        error,
        'Ledger: Error while signing EIP-7702 authorization',
      );
```

File: packages/keyring-eth-ledger-bridge/CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Added 7702 methods ([#564](https://github.com/MetaMask/accounts/pull/564))
```
Comment (gantunesr) [inline packages/keyring-eth-ledger-bridge/CHANGELOG.md]:
```suggestion
- Add EIP-7702 authorization signing to the Ledger keyring ([#564](https://github.com/MetaMask/accounts/pull/564))
    - Not yet functional: both iframe and mobile bridges throw until DMK support lands.
    - Fixes hex `v` parsing (`'1b'`) for personal_sign and typed data. 
```

File: packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts (hunk 1 above)
Comment (gantunesr) [reply packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts]: For `NaN` I would suggest throwing an error like: `Invalid recovery parameter from device`

Comment (gantunesr) [reply packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts]: And I would also suggest adding test cases that cover the following,

- Hex without prefix: '1b' → 27
- Hex with prefix: '0x1b' → 27
- String: '27' → 27
- Integer: 27 → 27
- Hex prefix: '0x' → throw

File: packages/keyring-eth-ledger-bridge/src/ledger-bridge.ts [hunk 6]
```diff
@@ -25,6 +25,17 @@ export type LedgerSignTypedDataResponse = Awaited<
...
+export type LedgerSignDelegationAuthorizationParams = {
+  hdPath: string;
+  chainId: number;
+  contractAddress: string;
+  nonce: number;
+};
+
+export type LedgerSignDelegationAuthorizationResponse = Awaited<
+  ReturnType<LedgerHwAppEth['signTransaction']>
+>;
```
Comment (gantunesr) [inline packages/keyring-eth-ledger-bridge/src/ledger-bridge.ts]: Is this a placeholder? Should it a delegation signature instead of `signTransaction`?

Comment (montelaidev) [reply packages/keyring-eth-ledger-bridge/src/ledger-bridge.ts]: Its the same return type. It just an object with rsv. We can just add a note to replace it

File: packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts (hunk 1 above)
Comment (montelaidev) [reply packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts]: Added additional tests

File: packages/keyring-eth-ledger-bridge/src/ledger-mobile-bridge.ts [hunk 7]
```diff
@@ -102,6 +104,17 @@ export class LedgerMobileBridge implements MobileBridge {
     return this.#getEthApp().signEIP712Message(hdPath, message);
   }
 
+  async deviceSignDelegationAuthorization({
+    hdPath: _hdPath,
+    chainId: _chainId,
+    contractAddress: _contractAddress,
+    nonce: _nonce,
+  }: LedgerSignDelegationAuthorizationParams): Promise<LedgerSignDelegationAuthorizationResponse> {
```
Comment (mathieuartu) [inline packages/keyring-eth-ledger-bridge/src/ledger-mobile-bridge.ts:112]: Nit: why destructure the arguments if we use none?

File: packages/keyring-eth-ledger-bridge/src/ledger-iframe-bridge.ts [hunk 8]
```diff
@@ -238,6 +240,14 @@ export class LedgerIframeBridge implements LedgerBridge<LedgerIframeBridgeOption
     );
   }
 
+  async deviceSignDelegationAuthorization(
+    _params: LedgerSignDelegationAuthorizationParams,
```
Comment (mathieuartu) [inline packages/keyring-eth-ledger-bridge/src/ledger-iframe-bridge.ts:244]: Naive Q: why optional `params` here and not elsewhere where `LedgerSignDelegationAuthorizationParams` is used in this PR?

File: packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts [hunk 9]
```diff
@@ -720,6 +774,38 @@ export class LedgerKeyring implements Keyring {
...
+    const value = String(recoveryParam).trim();
+
+    if (value.startsWith('0x') || value.startsWith('0X')) {
+      if (!isStrictHexString(value)) {
+        throw new Error(`Invalid hex recovery parameter: "${recoveryParam}"`);
+      }
+      return hexToNumber(value);
+    }
+
+    const radix = /[a-f]/iu.test(value) ? 16 : 10;
```
Comment (mathieuartu) [inline packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts:801]: Nit: would love a comment here explaining what this is

File: packages/keyring-eth-ledger-bridge/src/ledger-iframe-bridge.ts (hunk 8 above)
Comment (montelaidev) [reply packages/keyring-eth-ledger-bridge/src/ledger-iframe-bridge.ts:244]: The `hw-eth-app` doesn't support it. We will be introducing the DMK bridge that will be using it.

File: packages/keyring-eth-ledger-bridge/src/ledger-mobile-bridge.ts (hunk 7 above)
Comment (montelaidev) [reply packages/keyring-eth-ledger-bridge/src/ledger-mobile-bridge.ts:112]: The `hw-eth-app` doesn't support it. We will be introducing the DMK bridge that will be using it.

**Review by mathieuartu — APPROVED** [review]
LGTM, but might have missed something due to not being 100% up to date on the concept. Maybe wait for a second review before merging? But no issues on my end!

**Review by gantunesr — APPROVED** [review]
